router/router_test.go

```diff
@@ -74,6 +74,178 @@ func TestExchangeMap(t *testing.T) {
+			expectedBidderInfos: config.BidderInfos{"a": {Syncer: &config.Syncer{Supports: []string{"iframe"}, IFrame: &config.SyncerEndpoint{URL: "override"}}}},
+		},
+		{
+			description:         "UserSyncURL Override Redirect",
+			givenBidderInfos:    config.BidderInfos{"a": {Syncer: &config.Syncer{Supports: []string{"redirect"}}}},
+			givenAdaptersCfg:    map[string]config.Adapter{"a": {UserSyncURL: "override"}},
+			expectedBidderInfos: config.BidderInfos{"a": {Syncer: &config.Syncer{Supports: []string{"redirect"}, Redirect: &config.SyncerEndpoint{URL: "override"}}}},
+		},
+		{
+			description:         "UserSyncURL Supports Redirect",
+			givenBidderInfos:    config.BidderInfos{"a": {Syncer: &config.Syncer{Redirect: &config.SyncerEndpoint{URL: "original"}}}},
+			givenAdaptersCfg:    map[string]config.Adapter{"a": {UserSyncURL: "override"}},
+			expectedBidderInfos: config.BidderInfos{"a": {Syncer: &config.Syncer{Redirect: &config.SyncerEndpoint{URL: "override"}}}},
+		},
+		{
+			description:      "UserSyncURL Override Syncer Not Defined",
+			givenBidderInfos: config.BidderInfos{"a": {}},
+			givenAdaptersCfg: map[string]config.Adapter{"a": {UserSyncURL: "override"}},
+			expectedError:    "adapters.a.usersync_url cannot be applied, bidder does not define a user sync",
+		},
+		{
+			description:      "UserSyncURL Override Syncer Endpoints Not Defined",
+			givenBidderInfos: config.BidderInfos{"a": {Syncer: &config.Syncer{}}},
+			givenAdaptersCfg: map[string]config.Adapter{"a": {UserSyncURL: "override"}},
+			expectedError:    "adapters.a.usersync_url cannot be applied, bidder does not define user sync endpoints and does not define supported endpoints",
+		},
+		{
+			description:      "UserSyncURL Override Ambiguous",
+			givenBidderInfos: config.BidderInfos{"a": {Syncer: &config.Syncer{IFrame: &config.SyncerEndpoint{URL: "originalIFrame"}, Redirect: &config.SyncerEndpoint{URL: "originalRedirect"}}}},
+			givenAdaptersCfg: map[string]config.Adapter{"a": {UserSyncURL: "override"}},
+			expectedError:    "adapters.a.usersync_url cannot be applied, bidder defines multiple user sync endpoints or supports multiple endpoints",
+		},
+		{
+			description:      "UserSyncURL Supports Ambiguous",
+			givenBidderInfos: config.BidderInfos{"a": {Syncer: &config.Syncer{Supports: []string{"iframe", "redirect"}}}},
+			givenAdaptersCfg: map[string]config.Adapter{"a": {UserSyncURL: "override"}},
+			expectedError:    "adapters.a.usersync_url cannot be applied, bidder defines multiple user sync endpoints or supports multiple endpoints",
+		},
+	}
+
+	for _, test := range testCases {
+		resultErr := applyBidderInfoConfigOverrides(test.givenBidderInfos, test.givenAdaptersCfg)
+		if test.expectedError == "" {
+			assert.NoError(t, resultErr, test.description+":err")
+			assert.Equal(t, test.expectedBidderInfos, test.givenBidderInfos, test.description+":result")
+		} else {
+			assert.EqualError(t, resultErr, test.expectedError, test.description+":err")
+		}
```

Comment:
I realize that I'm not asserting the glog warnings. That's because we don't have a good way to intercept the log writes for tests. I'd like to see that fixed, but that would be too much work for this PR. I manually verified for now.